examples/ecs/ecs_guide.rs

```diff
@@ -285,11 +291,20 @@ fn main() {
         .add_stage_after(stage::UPDATE, "after_round", SystemStage::parallel())
         .add_system_to_stage("before_round", new_round_system.system())
         .add_system_to_stage("before_round", new_player_system.system())
-        .add_system_to_stage("after_round", score_check_system.system())
-        .add_system_to_stage("after_round", game_over_system.system())
-        // score_check_system will run before game_over_system because score_check_system modifies GameState and game_over_system
-        // reads GameState. This works, but it's a bit confusing. In practice, it would be clearer to create a new stage that runs
-        // before "after_round"
+        // We can ensure that game_over system runs after score_check_system using explicit ordering constraints
+        // First, we label the system we want to refer to using `.label`
+        // Then, we use either `.before` or `.after` to describe the order we want the relationship
+        .add_system_to_stage(
+            "after_round",
+            score_check_system.system().label("score_check"),
+        )
+        .add_system_to_stage(
+            "after_round",
+            game_over_system.system().after("score_check"),
+        )
+        // We can check our systems for execution order ambiguities by examining the output produced in the console
+        // by adding the following Resource to our App :)
+        .insert_resource(ReportExecutionOrderAmbiguities {})
```

Comment:
Not everything reported by the checker is a problem. I'm not sure if this should also mention upcoming ways to selectively silence the warnings.

`{}` are not needed for construction if the struct has no fields.

```suggestion
        // Be aware that not everything reported by this checker is a potential problem, you'll have to make
        // that judgement yourself.
        .insert_resource(ReportExecutionOrderAmbiguities)
```